radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/addressbook/AddressBookEntry.java

```diff
@@ -202,7 +210,33 @@ public AddressBookEntry withLastSuccessfulConnectionFor(RadixNodeUri uri, Instan
 				.build();
 			return new AddressBookEntry(nodeId, bannedUntil, newKnownAddresses);
 		} else {
-			final var newAddressEntry = new PeerAddressEntry(uri, Optional.of(lastSuccessfulConnectionFor));
+			final var newAddressEntry = new PeerAddressEntry(uri, Optional.of(lastSuccessfulConnectionFor), false);
+			final var newKnownAddresses = ImmutableSet.<PeerAddressEntry>builder()
+				.addAll(this.knownAddresses)
+				.add(newAddressEntry)
+				.build();
+			return new AddressBookEntry(nodeId, bannedUntil, newKnownAddresses);
+		}
+	}
+
+	public AddressBookEntry withBlacklistedUri(RadixNodeUri uri) {
```

Comment:
Do we have a retention policy for this growing URI deny list?

Reply:
Good point. I think we'd generally want to add a limit on address book entries stored.